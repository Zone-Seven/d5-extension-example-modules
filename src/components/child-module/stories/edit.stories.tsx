import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import {
  defaultAttributes,
  customIcon,
  iconStyle,
  titleStyle,
  contentStyle,
} from '../__mock-data__/attrs';

import { ChildModuleEdit } from '../edit';
import { ChildModuleEditProps } from '../types';
import { select } from '@divi/data';
import { childModule } from '..';
import { isNull, isUndefined, omit } from 'lodash';
import { registerModule, registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerModals } from '@divi/modal-library';
import { registerAppUiStore } from '@divi/app-ui';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSettingsStore } from '@divi/settings';
import { registerHistoryStore } from '@divi/history';
import { registerAppPreferencesStore } from '@divi/app-preferences';

// Create template to render argument given.
const templateDividerEdit: Story<ChildModuleEditProps> = args => createElement(ChildModuleEdit, args);

const name = 'example/child-module';

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'child-module-1',
  attrs: defaultAttributes,
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Child module is added.',
    },
  },
};

const CustomIcon      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
CustomIcon.args       = {
  name,
  id:    'child-module-2',
  attrs: customIcon,
};
CustomIcon.parameters = {
  docs: {
    description: {
      story: 'Display selected icon properly when the icon value is set.',
    },
  },
};

const IconStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
IconStyle.args       = {
  name,
  id:    'child-module-3',
  attrs: iconStyle,
};
IconStyle.parameters = {
  docs: {
    description: {
      story: 'Display icon style properly when the icon style is set.',
    },
  },
};

const TitleStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
TitleStyle.args       = {
  name,
  id:    'child-module-4',
  attrs: titleStyle,
};
TitleStyle.parameters = {
  docs: {
    description: {
      story: 'Display title style properly when the title style is set.',
    },
  },
};

const ContentStyle      = templateDividerEdit.bind({}) as Story<ChildModuleEditProps>;
ContentStyle.args       = {
  name,
  id:    'child-module-5',
  attrs: contentStyle,
};
ContentStyle.parameters = {
  docs: {
    description: {
      story: 'Display content style properly when the content style is set.',
    },
  },
};



export {
  DefaultValues,
  CustomIcon,
  IconStyle,
  TitleStyle,
  ContentStyle,
};

export default {
  title: 'Modules/Child Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerAppPreferencesStore();
      registerAppUiStore();
      registerEventsStore();
      registerKeyboardShortcutsStore();
      registerModals();
      registerSettingsStore();
      registerSerializedPostStore();
      registerEditPostStore();
      registerHistoryStore();
      if (isNull(select('divi/module-library'))) {
        registerModuleLibraryStore();
      }
      const theChildModule  = select('divi/module-library').getModule(childModule?.metadata?.name);

      if (isUndefined(theChildModule)) {
        registerModule(childModule.metadata, omit(childModule, 'metadata'));
      }
      return story();
    }),
  ],
};
