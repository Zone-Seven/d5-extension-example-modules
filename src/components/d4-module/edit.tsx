// External Dependencies.
import React, { ReactElement } from 'react';
import classnames from 'classnames';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { D4ModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';
import { ModuleScriptData } from './module-script-data';

/**
 * Divi 4 Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {D4ModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const D4ModuleEdit = (props: D4ModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const HeadingLevel = moduleAttrs?.titleFont?.font?.desktop?.value?.headingLevel;
  const title        = moduleAttrs?.title?.desktop?.value;
  const content      = moduleAttrs?.content?.desktop?.value;

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      scriptDataComponent={ModuleScriptData}
    >
      <HeadingLevel className="d4_module_title">{title}</HeadingLevel>
      <div
        className="d4_module_content"
        dangerouslySetInnerHTML={{ __html: content }}
      />
    </ModuleContainer>
  );
}

export {
  D4ModuleEdit,
};
