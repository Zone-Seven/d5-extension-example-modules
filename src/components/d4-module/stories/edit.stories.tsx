import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import {
  defaultAttributes,
  contentStyle,
  styleOne,
  styleTwo,
} from '../__mock-data__/attrs';

import { D4ModuleEdit } from '../edit';
import { D4ModuleEditProps } from '../types';
import { d4Module } from '..';
import { isNull, isUndefined, omit } from 'lodash';
import { registerModuleLibraryStore, registerModule } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerModalLibraryStore, registerModals } from '@divi/modal-library';
import { registerAppUiStore } from '@divi/app-ui';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSettingsStore } from '@divi/settings';
import { registerHistoryStore } from '@divi/history';
import { registerAppPreferencesStore } from '@divi/app-preferences';
import { select } from '@divi/data';

// Create template to render argument given.
const templateDividerEdit: Story<D4ModuleEditProps> = args => createElement(D4ModuleEdit, args);

const name = 'example/d4-module';

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<D4ModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'd4-module-1',
  attrs: defaultAttributes,
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Divi 4 module is added.',
    },
  },
};

const ContentStyle      = templateDividerEdit.bind({}) as Story<D4ModuleEditProps>;
ContentStyle.args       = {
  name,
  id:    'd4-module-2',
  attrs: contentStyle,
};
ContentStyle.parameters = {
  docs: {
    description: {
      story: 'Display content text style properly when the content text style is set.',
    },
  },
};

const StyleOne      = templateDividerEdit.bind({}) as Story<D4ModuleEditProps>;
StyleOne.args       = {
  name,
  id:    'd4-module-3',
  attrs: styleOne,
};
StyleOne.parameters = {
  docs: {
    description: {
      story: 'Display custom background, title color, content color and padding properly when the custom background, title color, content color and padding are set.',
    },
  },
};

const StyleTwo      = templateDividerEdit.bind({}) as Story<D4ModuleEditProps>;
StyleTwo.args       = {
  name,
  id:    'd4-module-4',
  attrs: styleTwo,
};
StyleTwo.parameters = {
  docs: {
    description: {
      story: 'Display custom styles properly when the custom styles are set.',
    },
  },
};


export {
  DefaultValues,
  ContentStyle,
  StyleOne,
  StyleTwo,
};

export default {
  title: 'Modules/Divi 4 Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerAppPreferencesStore();
      registerAppUiStore();
      registerEventsStore();
      registerKeyboardShortcutsStore();
      registerModals();
      registerSettingsStore();
      registerSerializedPostStore();
      registerEditPostStore();
      registerHistoryStore();
      if (isNull(select('divi/module-library'))) {
        registerModuleLibraryStore();
      }

      const theModule = select('divi/module-library').getModule(d4Module?.metadata?.name);

      if (isUndefined(theModule)) {
        registerModule(d4Module.metadata, omit(d4Module, 'metadata'));
      }
      return story();
    }),
  ],
};
