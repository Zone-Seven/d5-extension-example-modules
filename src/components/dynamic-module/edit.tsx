// External Dependencies.
import React, { ReactElement, useEffect } from 'react';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { DynamicModuleEditProps } from './types';
import { ModuleStyles } from './styles';
import { defaultAttrs } from './constants';
import { useGetRecentPosts } from './hooks/get-recent-posts';
import { map } from 'lodash';
import { __ } from '@wordpress/i18n';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {DynamicModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const DynamicModuleEdit = (props: DynamicModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const {getPosts, posts, isLoading} = useGetRecentPosts();

  const title         = moduleAttrs?.title?.desktop?.value;
  const numberOfPosts = parseInt(moduleAttrs?.numberOfPosts?.desktop?.value);
  const TitleHeading  = moduleAttrs?.titleFont?.font?.desktop?.value?.headingLevel;
  const PostTitleHeading  = moduleAttrs?.postTitleFont?.font?.desktop?.value?.headingLevel;

  /**
   * Fetches new Portfolio Posts on parameter changes.
   */
   useEffect(() => {
    getPosts(numberOfPosts);
  }, [numberOfPosts]);

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
    >
      {
        ! isLoading && (
          <>
            {
              title && (
                <TitleHeading className="et_pb_dynamic_module_title">{title}</TitleHeading>
              )
            }
            <div className="et_pb_dynamic_module_post_items">
              {
                map(posts, (post) => (
                    <div className="et_pb_dynamic_module_post_item">
                      <PostTitleHeading className="et_pb_dynamic_module_post_item_title">
                        <a href={post?.link} onClick={() => false}>{post?.title?.rendered}</a>
                      </PostTitleHeading>
                      <div className="et_pb_dynamic_module_post_item_content" dangerouslySetInnerHTML={{__html: post?.excerpt?.rendered}} />
                    </div>
                  )
                )
              }
            </div>
          </>
        )
      }
      {
        ! isLoading && posts.length < 1 && (
          <div>{__('No post found.', 'd5-module-extension-example')}</div>
        )
      }
      {
        isLoading && (
          <div>{__('Loading...', 'd5-module-extension-example')}</div>
        )
      }
    </ModuleContainer>
  );
}

export {
  DynamicModuleEdit,
};
