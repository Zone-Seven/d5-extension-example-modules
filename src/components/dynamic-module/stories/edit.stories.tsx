import {
  createElement,
  ReactElement,
} from 'react';
/* eslint-disable-next-line import/no-extraneous-dependencies */
import { Story } from '@storybook/react';
import { rest } from 'msw';

import {
  defaultAttributes, numberOfPosts, postTitleStyle, titleStyle,
} from '../__mock-data__/attrs';

import { DynamicModuleEdit } from '../edit';
import { DynamicModuleEditProps } from '../types';
import { select } from '@divi/data';
import { registerEditPostStore } from '@divi/edit-post';
import { dynamicModule } from '..';
import { isNull, isUndefined, omit, times } from 'lodash';
import { registerModule, registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerModals } from '@divi/modal-library';
import { registerAppUiStore } from '@divi/app-ui';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { registerSettingsStore } from '@divi/settings';
import { registerHistoryStore } from '@divi/history';
import { registerAppPreferencesStore } from '@divi/app-preferences';
import { registerAjaxStore } from '@divi/ajax';
import { post } from '../__mock-data__/posts';

// Create template to render argument given.
const templateDividerEdit: Story<DynamicModuleEditProps> = args => createElement(DynamicModuleEdit, args);

const name = 'example/dynamic-module';

const posts = rest.get('wp-json/wp/v2/posts', (req, res, ctx) => {
  const perPage = req?.url?.searchParams?.get('per_page');
  
  return res(        
    ctx.json(times(parseInt(perPage), () => post)),
  );
});

// Stories.
const DefaultValues      = templateDividerEdit.bind({}) as Story<DynamicModuleEditProps>;
DefaultValues.args       = {
  name,
  id:    'dynamicModule1',
  attrs: defaultAttributes,
};
DefaultValues.parameters = {
  docs: {
    description: {
      story: 'Default attributes that gets added when Dynamic module is added.',
    },
  },
  msw: {
    handlers: {
      posts,
    },
  },
};

const NumberOfPosts      = templateDividerEdit.bind({}) as Story<DynamicModuleEditProps>;
NumberOfPosts.args       = {
  name,
  id:    'dynamicModule2',
  attrs: numberOfPosts,
};
NumberOfPosts.parameters = {
  docs: {
    description: {
      story: 'Display only 2 posts when the numberOfPosts attribute is set to 2.',
    },
  },
  msw: {
    handlers: {
      posts,
    },
  },
};

const TitleStyle      = templateDividerEdit.bind({}) as Story<DynamicModuleEditProps>;
TitleStyle.args       = {
  name,
  id:    'dynamicModule3',
  attrs: titleStyle,
};
TitleStyle.parameters = {
  docs: {
    description: {
      story: 'Display title style properly when the title style is set.',
    },
  },
  msw: {
    handlers: {
      posts,
    },
  },
};

const PostTitleStyle      = templateDividerEdit.bind({}) as Story<DynamicModuleEditProps>;
PostTitleStyle.args       = {
  name,
  id:    'dynamicModule3',
  attrs: postTitleStyle,
};
PostTitleStyle.parameters = {
  docs: {
    description: {
      story: 'Display title style properly when the post title style is set.',
    },
  },
  msw: {
    handlers: {
      posts,
    },
  },
};


export {
  DefaultValues,
  NumberOfPosts,
  TitleStyle,
  PostTitleStyle,
};

export default {
  title: 'Modules/Dynamic Module/Edit',
  decorators: [
    ((story: () => ReactElement): ReactElement => {
      registerAppPreferencesStore();
      registerAppUiStore();
      registerEventsStore();
      registerKeyboardShortcutsStore();
      registerModals();
      registerSettingsStore();
      registerAjaxStore();
      registerSerializedPostStore();
      registerEditPostStore();
      registerHistoryStore();
      if (isNull(select('divi/module-library'))) {
        registerModuleLibraryStore();
      }

      const theModule  = select('divi/module-library').getModule(dynamicModule?.metadata?.name);

      if (isUndefined(theModule)) {
        registerModule(dynamicModule.metadata, omit(dynamicModule, 'metadata'));
      }
      return story();
    }),
  ],
};
