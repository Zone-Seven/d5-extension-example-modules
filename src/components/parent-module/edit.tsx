// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ChildModulesContainer, ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';

// Local Dependencies.
import { ParentModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { ModuleStyles } from './styles';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ParentModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ParentModuleEdit = (props: ParentModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    childrenIds,
  } = props;

  // Merge module default values with module attributes.
  const moduleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      tag="ul"
    >
      <ChildModulesContainer ids={childrenIds} parentId={id} />
    </ModuleContainer>
  );
}

export {
  ParentModuleEdit,
};
