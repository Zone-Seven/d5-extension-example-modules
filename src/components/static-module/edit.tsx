// External Dependencies.
import React, { ReactElement } from 'react';
import classnames from 'classnames';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';

// Local Dependencies.
import { StaticModuleEditProps } from './types';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';
import { ModuleScriptData } from './module-script-data';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {StaticModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const StaticModuleEdit = (props: StaticModuleEditProps): ReactElement => {
  const {
    attrs,
    elements,
    id,
    name,
  } = props;

  // Get module attributes.
  const imageSrc                = attrs?.image?.innerContent?.desktop?.value?.src;
  const imageAlt                = attrs?.image?.innerContent?.desktop?.value?.alt;

  return (
    <ModuleContainer
      attrs={attrs}
      elements={elements}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      scriptDataComponent={ModuleScriptData}
    >
      <div className="static-module__image">
        <img src={imageSrc} alt={imageAlt} />
      </div>
      <div className="static-module__content-container">
        {elements.render({
          attrName: 'title',
        })}
        <div className="static-module__content">
          {elements.render({
            attrName: 'content',
          })}
        </div>
      </div>
    </ModuleContainer>
  );
}

export {
  StaticModuleEdit,
};
