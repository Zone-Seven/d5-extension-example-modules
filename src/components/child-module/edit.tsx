// External Dependencies.
import React, { ReactElement } from 'react';

// Divi Dependencies.
import { ModuleContainer } from '@divi/module';
import { mergeAttrs } from '@divi/module-utils';
import { processFontIcon } from '@divi/icon-library';

// Local Dependencies.
import { ChildModuleEditProps } from './types';
import { defaultAttrs } from './constants';
import { defaultAttrs as parentDefaultAttrs } from '../parent-module/constants';
import { ModuleStyles } from './styles';
import { moduleClassnames } from './module-classnames';

/**
 * Static Module edit component of visual builder.
 *
 * @since ??
 *
 * @param {ChildModuleEditProps} props React component props.
 *
 * @returns {ReactElement}
 */
const ChildModuleEdit = (props: ChildModuleEditProps): ReactElement => {
  const {
    attrs,
    id,
    name,
    parentAttrs,
  } = props;

  // Merge parent module default values with parent module attributes.
  const parentModuleAttrs = mergeAttrs({
    defaultAttrs: parentDefaultAttrs,
    attrs: parentAttrs,
  });

  // Merge module default values with module attributes.
  const childModuleAttrs = mergeAttrs({
    defaultAttrs,
    attrs,
  });

  const moduleAttrs = {
    ...childModuleAttrs,
    icon: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.icon,
      attrs: childModuleAttrs?.icon,
    }),
    iconColor: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconColor,
      attrs: childModuleAttrs?.iconColor,
    }),
    iconSize: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.iconSize,
      attrs: childModuleAttrs?.iconSize,
    }),
    titleFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.titleFont,
      attrs: childModuleAttrs?.titleFont,
    }),
    text: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.text,
      attrs: childModuleAttrs?.text,
    }),
    bodyFont: mergeAttrs({
      defaultAttrs: parentModuleAttrs?.bodyFont,
      attrs: childModuleAttrs?.bodyFont,
    }),
  };
  
  const icon    = moduleAttrs?.icon?.desktop?.value;
  const title   = moduleAttrs?.title?.desktop?.value;
  const content = moduleAttrs?.content?.desktop?.value;

  return (
    <ModuleContainer
      attrs={moduleAttrs}
      componentType="edit"
      id={id}
      name={name}
      stylesComponent={ModuleStyles}
      classnamesFunction={moduleClassnames}
      tag="li"
    >
      { icon && (
        <div className="child-module__icon et-pb-icon">
            {processFontIcon(icon)}
        </div>
      )}
      <div className="child-module__content-container">
        <div className="child-module__title">{title}</div>
        <div
          className="child-module__content"
          dangerouslySetInnerHTML={{ __html: content }}
        />
      </div>
    </ModuleContainer>
  );
}

export {
  ChildModuleEdit,
};
