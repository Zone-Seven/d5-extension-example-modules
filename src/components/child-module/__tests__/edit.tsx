import { createElement } from 'react';
import { shallow } from 'enzyme';
import {
  forEach,
  pickBy,
  has,
  get,
} from 'lodash';

import { registerAppUiStore } from '@divi/app-ui';
import { registerModuleLibraryStore } from '@divi/module-library';
import { registerEventsStore } from '@divi/events';
import { registerKeyboardShortcutsStore } from '@divi/keyboard-shortcuts';
import { registerModals } from '@divi/modal-library';
import { registerEditPostStore } from '@divi/edit-post';
import { registerSerializedPostStore } from '@divi/serialized-post';
import { ChildModuleEdit } from '../edit';
import { dispatch } from '@divi/data';
import { childModule } from '..';

import * as editStories  from '../stories/edit.stories';
import { registerAppPreferencesStore } from '@divi/app-preferences';

// Only test story that has description. This naturally excludes `default` export from being included as test.
const testableStories = pickBy(editStories, story => has(story, 'parameters.docs.description.story')) as Omit<typeof editStories, 'default'>;

beforeAll(() => {
  registerModuleLibraryStore();
  dispatch('divi/module-library').addModule(childModule);
});

beforeEach(() => {
  registerAppPreferencesStore();
  registerAppUiStore();
  registerEventsStore();
  registerKeyboardShortcutsStore();
  registerModals();
  registerSerializedPostStore();
  registerEditPostStore();
});

describe('<ChildModuleEdit />', () => {
  // Snapshot test based on storybook configuration.
  // Prevents regression due to unintentionally modifying component.
  forEach(testableStories, testableStory => {
    it(`Should match the snapshot when "${get(testableStory, 'parameters.docs.description.story') as string}"`, () => {
      expect(
        shallow(createElement(ChildModuleEdit, testableStory.args)).html(),
      ).toMatchSnapshot();
    });
  });
});
