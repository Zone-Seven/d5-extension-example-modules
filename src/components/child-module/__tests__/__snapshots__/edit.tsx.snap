// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ChildModuleEdit /> Should match the snapshot when "Default attributes that gets added when Child module is added." 1`] = `"<li class=\\"example_child_module_child-module-1 example_child_module et_pb_module et-animated--vb\\" data-id=\\"child-module-1\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\" data-et-vb-editable-element=\\"false\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><div id=\\"tiny_mce_child-module-1\\" class=\\"child-module__content\\" data-et-vb-popover-tinymce=\\"false\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div></li>"`;

exports[`<ChildModuleEdit /> Should match the snapshot when "Display content style properly when the content style is set." 1`] = `"<li class=\\"example_child_module_child-module-5 example_child_module et_pb_module et-animated--vb\\" data-id=\\"child-module-5\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\" data-et-vb-editable-element=\\"false\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><div id=\\"tiny_mce_child-module-5\\" class=\\"child-module__content\\" data-et-vb-popover-tinymce=\\"false\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div><style class=\\"et-vb-module-styles\\">.parent-module .example_child_module_child-module-5 .child-module__content {font-variant: small-caps; text-decoration-line: underline; text-decoration-style: solid; color: #ae16f0; font-size: 16px;}</style></li>"`;

exports[`<ChildModuleEdit /> Should match the snapshot when "Display icon style properly when the icon style is set." 1`] = `"<li class=\\"example_child_module_child-module-3 example_child_module et_pb_module et-animated--vb\\" data-id=\\"child-module-3\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\" data-et-vb-editable-element=\\"false\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><div id=\\"tiny_mce_child-module-3\\" class=\\"child-module__content\\" data-et-vb-popover-tinymce=\\"false\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div><style class=\\"et-vb-module-styles\\">.example_child_module_child-module-3 .child-module__icon.et-pb-icon {content: &#x27;9&#x27; !important; font-family: &quot;ETmodules&quot; !important;}.example_child_module_child-module-3 .child-module__icon.et-pb-icon {color: #ae16f0;}.example_child_module_child-module-3 .child-module__icon.et-pb-icon {font-size: 28px;}</style></li>"`;

exports[`<ChildModuleEdit /> Should match the snapshot when "Display selected icon properly when the icon value is set." 1`] = `"<li class=\\"example_child_module_child-module-2 example_child_module et_pb_module et-animated--vb\\" data-id=\\"child-module-2\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\" data-et-vb-editable-element=\\"false\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><div id=\\"tiny_mce_child-module-2\\" class=\\"child-module__content\\" data-et-vb-popover-tinymce=\\"false\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div><style class=\\"et-vb-module-styles\\">.example_child_module_child-module-2 .child-module__icon.et-pb-icon {content: &#x27;9&#x27; !important; font-family: &quot;ETmodules&quot; !important;}</style></li>"`;

exports[`<ChildModuleEdit /> Should match the snapshot when "Display title style properly when the title style is set." 1`] = `"<li class=\\"example_child_module_child-module-4 example_child_module et_pb_module et-animated--vb\\" data-id=\\"child-module-4\\"><div class=\\"child-module__icon et-pb-icon\\">9</div><div class=\\"child-module__content-container\\"><div class=\\"child-module__title\\" data-et-vb-editable-element=\\"false\\">Your Title Goes Here</div><div class=\\"child-module__content\\"><div id=\\"tiny_mce_child-module-4\\" class=\\"child-module__content\\" data-et-vb-popover-tinymce=\\"false\\"><p>Your content goes here. Edit or remove this text inline or in the module Content settings. You can also style every aspect of this content in the module Design settings and even apply custom CSS to this text in the module Advanced settings.</p></div></div></div><style class=\\"et-vb-module-styles\\">.parent-module .example_child_module_child-module-4 .child-module__title {font-weight: 300; font-style: italic; text-transform: uppercase; color: #ae16f0 !important; font-size: 31px;}</style></li>"`;
